src/lib.rs

```diff
@@ -1,287 +1,284 @@
-pub mod blockstore;
 pub mod ext;
 pub mod state;
 pub mod types;
 mod utils;
 
-use anyhow::anyhow;
-use cid::Cid;
-use ext::sca::SCA_ACTOR_ADDR;
-use fil_actor_hierarchical_sca::{Checkpoint, FundParams, Method, MIN_COLLATERAL_AMOUNT};
-use fvm_ipld_encoding::{RawBytes, DAG_CBOR};
-use fvm_sdk as sdk;
-use fvm_sdk::NO_DATA_BLOCK_ID;
-use fvm_shared::actor::builtin::Type;
-use fvm_shared::address::Address;
-use fvm_shared::bigint::bigint_ser::BigIntDe;
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::{actor_error, cbor, ActorDowncast, ActorError, INIT_ACTOR_ADDR};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::RawBytes;
+
 use fvm_shared::econ::TokenAmount;
-use fvm_shared::{ActorID, METHOD_SEND};
-use num_traits::Zero;
-use sdk::actor::get_actor_code_cid;
-use state::get_votes;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use ipc_gateway::MIN_COLLATERAL_AMOUNT;
+use num_derive::FromPrimitive;
+use num_traits::{FromPrimitive, Zero};
 
-use crate::blockstore::*;
-use crate::state::{get_stake, State};
+use crate::state::State;
 use crate::types::*;
 use crate::utils::*;
 
-pub const TEST_ADDR_ID: ActorID = 339;
-
-/// The actor's WASM entrypoint. It takes the ID of the parameters block,
-/// and returns the ID of the return value block, or NO_DATA_BLOCK_ID if no
-/// return value.
-///
-/// Should probably have macros similar to the ones on fvm.filecoin.io snippets.
-/// Put all methods inside an impl struct and annotate it with a derive macro
-/// that handles state serde and dispatch.
-#[no_mangle]
-pub fn invoke(params: u32) -> u32 {
-    let params = sdk::message::params_raw(params).unwrap().1;
-    let params = RawBytes::new(params);
-    // Conduct method dispatch. Handle input parameters and return data.
-    let ret: anyhow::Result<Option<RawBytes>> = match sdk::message::method_number() {
-        1 => Actor::constructor(deserialize_params(&params).unwrap()),
-        2 => Actor::join(deserialize_params(&params).unwrap()),
-        3 => Actor::leave(),
-        4 => Actor::kill(),
-        5 => Actor::submit_checkpoint(deserialize_params(&params).unwrap()),
-        _ => abort!(USR_UNHANDLED_MESSAGE, "unrecognized method"),
-    };
+fil_actors_runtime::wasm_trampoline!(Actor);
```

Comment:
The trampoline should be part of the `fvm_utils` and our implementation of the runtime, if this is not the case we should make the change and propagate this to all of our actors.

Reply:
Also my bad, I thought the runtime was being pulled from `ref-fvm`.